pkg/volume/volume_info_common.go

```diff
@@ -0,0 +1,80 @@
+package volume
+
+type VolumeInfoVersion struct {
+	Version string `json:"version"`
+}
+
+const (
+	VolumeInfoVersionV1 string = "1"
+)
+
+type VolumeInfos struct {
+	VolumeInfosV1 []VolumeInfoV1
+	Version       string
+}
+
+// CSISnapshotInfo is used for displaying the CSI snapshot status
+type CSISnapshotInfo struct {
+	// The actual snapshot ID. It's the storage provider's snapshot ID for CSI.
+	SnapshotHandle string `json:"snapshotHandle"`
+
+	// The snapshot corresponding volume size. Some of the volume backup methods cannot retrieve the data by current design, for example, the Velero native snapshot.
+	Size int64 `json:"size"`
+
+	// The name of the CSI driver.
+	Driver string `json:"driver"`
+
+	// The name of the VolumeSnapshotContent.
+	VSCName string `json:"vscName"`
+}
+
+// SnapshotDataMoveInfo is used for displaying the snapshot data mover status.
+type SnapshotDataMoveInfo struct {
+	// The data mover used by the backup. The valid values are `velero` and ``(equals to `velero`).
+	DataMover string `json:"dataMover"`
+
+	// The type of the uploader that uploads the snapshot data. The valid values are `kopia` and `restic`. It's useful for file-system backup and snapshot data mover.
+	UploaderType string `json:"uploaderType"`
+
+	// The name or ID of the snapshot associated object(SAO).
```

Comment:
Just curious. What exactly is SAO? If you can give an example, that would be great.

Reply:
That's my bad.
SAO is a concept taken from an in-progress PR to support local snapshots for the snapshot data mover.
https://github.com/vmware-tanzu/velero/pull/7002/files#diff-e6456a41611c6ae9aa199c7cba18d13244ae635abaf7c63f706ace5ab956812eR661

Reply:
Thanks, will check that PR.